jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainActivity.java

```diff
@@ -96,11 +118,37 @@ public void sendMessage(String message) {
         // If the display message fragment is visible show the message there.
         if (displayMessageFragment != null) {
             displayMessageFragment.updateMessage(message);
-            return;
+        } else {
+            Log.w(MAIN_ACTIVITY_TAG,
+                    "Trying to send message, but displayMessageFragment is null.");
+        }
+
+        if (messageFile == null) {
```

Comment:
Actually, shouldn't it be 
```javas
if (messageFile.exists()) {
...
}
```
or something similar?
We don't actually use the `messageFile` reference in the `writeMessageToFile` method, we just want to make sure the file is actually created before we write to it.